### DISCUSSION TITLE
How can I use Husky on Windows and Mac?

### DISCUSSION THREAD

**Discussion by ntulip-qpac** [alirezanet/Husky.Net#62] [Q&A] [answered]
Hi,

Could someone point me to an example of how to configure Husky to run on Windows and Mac? I have it running on my mac just fine but we have other developers who use Windows and for them Husky is skipped.

I used [this article](https://blog.antosubash.com/posts/automatic-version-and-release#creating-github-action) as a reference. I've attached Husky to all the projects in the solution. 

I only use one pre-commit hook which runs a CSX script that forces conventional commit messages.

Thank you

**Comment by alirezanet** [comment]
Hi @ntulip-qpac,

Can you share your task-runner.json and commit-msg hook configuration?

**Reply by ntulip-qpac** [reply]
Sure, here they are attached. I build this while on a mac. 

Task Runner
````
{
   "tasks": [
      {
         "name": "dotnet-format",
         "group": "pre-commit",
         "command": "dotnet",
         "args": [ "dotnet-format", "--include", "${staged}" ],
         "include": [ "**/*.cs", "**/*.vb" ]
      },
      {
         "name": "commit-message-linter",
         "group": "pre-commit",
         "command": "dotnet",
         "args": [
            "husky",
            "exec",
            ".husky/csx/commit-lint.csx",
            "--args",
            "${args}"
         ]
      }
   ]
}

````

commit-msg

````
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

dotnet husky run --name "commit-message-linter" --args "$1"
````

**Reply by alirezanet** [reply]
Looks correct to me, 
and What is the error message ?!

**Reply by alirezanet** [reply]
also, if you add -v for verbose logs to the commands, it should give you more information

**Reply by ntulip-qpac** [reply]
We were testing the hooks and attempting to push a commit without the conventional message and were expecting to get an error from Husky. However, the commit went through.

`git commit -m "test"`

Our goal was to check that the commit message subject was following the pattern <type>(optional): <subject>

**Reply by alirezanet** [reply]
I'm sorry for not getting back to you sooner; I've tested the same configuration and it was working fine for me on Linux and windows, and unfortunately, I don't have access to Mac at the moment to test it out. 

Can you test your exec command directly from the command line? without hook involvement. 

-1 Create a text file `echo "test commit message" > /PATH/test.txt`
-2 run `dotnet husky exec PATH-TO-CSX --args PATH-TO-test.txt`

this should trigger your commit-message-linter and give you an invalid commit message error.

if it works, the problem probably is related to your git configuration.

**Reply by alirezanet** [reply]
can you also share your commit-message-linter.csx file? are you using the same file in this repo?

**Reply by KienMonoMono** [reply]
Hi
I'm using mac and try to  show all args but empty
foreach (var arg in Args)
{
    Console.WriteLine($"arg: {arg}");
}

**Reply by KienMonoMono** [reply]
I'm using validate script then show error: "Unable to execute cached script."
[
<img width="1632" alt="Screen Shot 2023-02-07 at 19 07 03" src="https://user-images.githubusercontent.com/100674200/217240769-8fa3b5cc-4f17-48a1-bd2a-a94046b9d8ca.png">
](url)

**Reply by KienMonoMono** [reply]
> I'm using validate script then show error: "Unable to execute cached script." [ <img alt="Screen Shot 2023-02-07 at 19 07 03" width="1632" src="https://user-images.githubusercontent.com/100674200/217240769-8fa3b5cc-4f17-48a1-bd2a-a94046b9d8ca.png"> ](url)

@alirezanet can you help me

**Reply by alirezanet** [reply, marked as answer]
Hi @KienMonoMono,
are you passing the arg correctly? 

commit-msg
```
dotnet husky run --name "commit-message-linter" --args "$1"
```

task-runner.json
```
   "tasks": [
      {
         "name": "commit-message-linter",
         "command": "dotnet",
         "args": ["husky", "exec", ".husky/csx/commit-lint.csx", "--args", "${args}"]
      },
...
```
